components/script/script_thread.rs

```diff
@@ -1489,35 +1494,51 @@ impl ScriptThread {
         }
     }
 
-    pub fn queue_run_job(&self, run_job_handler: Box<RunJobHandler>, global: &GlobalScope) {
+    pub fn queue_run_job(&self, run_job_handler: Box<AsyncJobHandler>, global: &GlobalScope) {
+        let _ = self.dom_manipulation_task_source.queue(run_job_handler, global);
+    }
+
+    pub fn invoke_run_job(&self, run_job_handler: Box<AsyncJobHandler>) {
+        let job_queue = &*self.job_queue_map;
+        job_queue.run_job(run_job_handler, self);
+    }
+
+    pub fn queue_register_job(&self, run_job_handler: Box<AsyncJobHandler>) {
```

Comment:
Looks like we can have a single `queue_service_worker_job` method, since queue_run_job, queue_async_job and _queue_register job are virtually identical.